### ISSUE TITLE
Any real world usage example I can use to inspire myself?

### ISSUE THREAD

**Issue by Rydra** [sartography/SpiffWorkflow#86]
Hi,

I've been taking a look at Spiffworkflow for a project I'm working on and it seems it could perfectly fit the bill, since I need to develop a workflow system where a process may have several steps, several people are involved in this process and have different permissions to execute certain steps, etc.

However I'm having trouble finding real usages of this library, and how to bind each task to python code or functions (e.g. notify by email a user once a step has been completed), how to decide which person is allowed to execute a certain step on a workflow, etc. Is there any complex implementation I could see as a source of inspiration?

Thank you and kind regards.

**Comment by Rydra** [comment]
@GCdeepvoice @dbutlerdb I've had to fork the project and make some small modifications to make it work to my needs, but the project is great for any business or data project that may require task orchestration (and ofc, python).

Can't show any code due to personal reasons, but on a small project of mine where I required to execute tasks in a specific order and have some conditional branches to execute some task or another depending on a value, I subclassed the SimpleTask and ExclusiveTask (and create serializers for it as explained in the documentation) and attach a function to those tasks (the event system that spiff implements is not enough for me, I had to expand it). Once I had the infrastructure ready, I could create complex workflows quite effortlessly through code (I didn't try to create workflows with json, it's too complex).

So TL;DR: in a world dominated by Java, spiff was the only pure python workflow engine that did the job I needed (and without adding any hard dependencies like django or twisted, giving me freedom). Having said that, the documentation is lackluster and you need to skim the source code over and over to understand every bit of it. It's not very complex though.

Hope it helps!

**Comment by rSrkn** [comment]
@Rydra Is it possible to create below workflow with SpiffWorkflow? 
- A user creates a request more resources from a form (Fields number of CPUs , Memory they request.) .
- Notification email sent to Management user. (Email will have the text of the form field.)
- Management user visit list of requests page (List will have all the details of the requests including timestamp)
- Management user approve or reject
- Initial user receive notification email about approval / rejection.
...

**Comment by dickkniep** [comment]
We at TM7 in the Netherlands developed an NLP engine that is usable in the real world i.e. government, banking etc. We use Camunda modeler to model the process and have developed many BPMN modules to perform various tasks. Spiffworkflow works nicely and gives us enough flexibility to perform the different tasks.

**Comment by calexh-sar** [comment]
@Rydra, @daniel-butlerm @CGdeepvoice and @dickkniep I am with Sartography and we just released [Version 1.0](https://github.com/sartography/SpiffWorkflow/releases/tag/1.0) which includes updates to the [readme docs](https://spiffworkflow.readthedocs.io/en/latest/ if you are still interested.

**Comment by calexh-sar** [comment]
Closing this issue due to age and lack of response.
